src/ucp/rndv/rndv_rkey_ptr.c

```diff
@@ -135,8 +136,15 @@ static unsigned ucp_proto_rndv_progress_rkey_ptr(void *arg)
     ucp_trace_req(req, "rkey_ptr unpack %zd from %p at offset %zd/%zd",
                   seg_size, src, offset, length);
 
+#ifdef ENABLE_AMD_BUFFER_TRANSFER
```

Comment:
can we avoid the compile-time option, and select the memcpy implementation according to the runtime CPU detection?
see https://github.com/openucx/ucx/blob/2d6595440b124df0631d7ef60f6bbde036ef55db/src/ucs/arch/x86_64/global_opts.h#L23-L23 as example

Reply:
@yosefe Thanks for the review. In __amd_buffer_transfer_method__ I need to identify the direction of memory transfer, I explained it in the attached figure (shared_memory_transfer.jpg) for the copy-in-copy-out case.  Is this way of identifying the transfer direction acceptable? or Can I get it as an input in ucs_memcpy_relaxed(void *dst, const void *src, size_t len, int direction)?
So in short this is not just another version of memcpy but rather a direction-aware variant. Will other architectures be interested in this method?


![shared_memory_transfer](https://github.com/openucx/ucx/assets/146521248/daa3777a-8316-48c8-9e0e-de2c8768dec7)

Reply:
Hi @arun-chandran-amd , we can definitely extend ucs_memcpy_relaxed to get more parameters that could improve performance, even if some architectures do not benefit from it.

Reply:
@yosefe Do you have any ideas/pointers on how to implement this? Can multiple calls to ucp_tag_send_xxx from the threads inside a process/rank will result in simultaneous calls to  ucs_memcpy_relaxed() in the sender/receiver?